devlib/utils/ssh.py

```diff
@@ -727,11 +727,11 @@ def _gem5_shell(self, command, as_root=False, timeout=None, check_exit_code=True
 
         output = output[command_index + len(command):].strip()
 
-        # It is possible that gem5 will echo the command. Therefore, we need to
-        # remove that too!
-        command_index = output.find(command)
-        if command_index != -1:
-            output = output[command_index + len(command):].strip()
+        # If gem5 echoes the command we need to remove that too!
+        if self.shell_echo:
+            command_index = output.find(command)
+            if command_index != -1:
+                output = output[command_index + len(command):].strip()
```

Comment:
Do we need to do something more here to make sure that this works as intended? We might want to limit for far it is allowed to look into the output, to try and avoid it matching things in the middle of the output. Just a thought - maybe I am thinking too much into this.

Reply:
I don't think it's worth changing anything here, since we have no means of testing the change would actually help. I think we need to wait until this becomes an issue.